### PR TITLE
WR324448: Add seb cipher for encrypting and decrypting seb config files

### PR DESCRIPTION
**Pull request by andrewmadden**

### CODE REVIEW

File: classes/seb_cipher.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+        foreach ($replacements as $key => $replacement) {
+            $xml = str_replace($key, $replacement, $xml);
+        }
+
+        // Stripe the trailing newline.
+        $data = trim($xml);
+
+        // First pass compression of encrypted data.
+        $data = gzencode($data);
```
Comment (dmitriim) [inline classes/seb_cipher.php:78]: Not sure if I understand why we do that

File: classes/seb_cipher.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        // Encrypt the data, return binary.
+        $data = $cryptor->encrypt($data, $password, Cryptor::DEFAULT_SCHEMA_VERSION, false);
+
+        // Add the magic header for password encrypted SEB files.
+        $data = 'pswd' . $data;
```
Comment (dmitriim) [inline classes/seb_cipher.php]: why we do that?

File: classes/seb_cipher.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    public static function decrypt(string $data, string $password = '') : string {
+        $cryptor = new Decryptor;
+
+        // If the file starts with <?xml then we really shouldn't be decrypting it.
+        if (strpos($data, '<?xml') === 0) {
+            return $data;
+        }
+
+        // First pass decompress to remove gz header. Suppress warning so false is returned if decode fails.
+        $data = @gzdecode($data);
```
Comment (dmitriim) [inline classes/seb_cipher.php]: Is there any other way to avoid using @?

File: classes/seb_cipher.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        // Check to see if magic header is 'pswd' which tells us it's password encrypted.
+        if (strpos($data, 'pswd') !== 0) {
+            return '';
+        }
+
+        // Remove the header.
```
Comment (dmitriim) [inline classes/seb_cipher.php]: What header? Why 4 ?

File: classes/seb_cipher.php (hunk 1 above)
Comment (andrewmadden) [reply classes/seb_cipher.php:78]: I've added detailed descriptions of process in method docs and references to steps in code comments.

File: classes/seb_cipher.php (hunk 3 above)
Comment (andrewmadden) [reply classes/seb_cipher.php]: I've removed warning suppressions and changed file_manager tests to handle the warning being thrown.

**Pull request merged**